File: Dsa-day93.py
class Solution:
    def maxDistinctElements(self, nums: List[int], k: int) -> int:
        nums.sort()
        prev = -float('inf')
        count = 0
        for x in nums:
            low, high = x - k, x + k
            val = max(prev + 1, low)
            if val <= high:
                count += 1
                prev = val
        return count

File: Dsa-day93.java
class Solution {
    public int maxDistinctElements(int[] nums, int k) {
        Arrays.sort(nums);
        long prev = Long.MIN_VALUE;
        int count = 0;
        for (int x : nums) {
            long low = (long)x - k, high = (long)x + k;
            long val = Math.max(prev + 1, low);
            if (val <= high) {
                count++;
                prev = val;
            }
        }
        return count;
    }
}
